api/server/router/local/image.go

```diff
@@ -184,12 +177,7 @@ func (s *router) postImagesPush(ctx context.Context, w http.ResponseWriter, r *h
 
 	name := vars["name"]
 	output := ioutils.NewWriteFlusher(w)
-	imagePushConfig := &graph.ImagePushConfig{
-		MetaHeaders: metaHeaders,
-		AuthConfig:  authConfig,
-		Tag:         r.Form.Get("tag"),
-		OutStream:   output,
-	}
+	imagePushConfig := graph.NewImagePushConfig(metaHeaders, authConfig, r.Form.Get("tag"), output)
```

Comment:
Yeah, I mean that `graph.NewImagePushConfig` for me personally made code less readable(actually almost unreadable) because whole point of structs was to not guess parameters positions.